src/main/scala/viper/silver/ast/utility/QuantifiedPermissions.scala

```diff
@@ -244,6 +244,19 @@ object QuantifiedPermissions {
 
             desugarSourceQuantifiedPermissionSyntax(Forall(vars ++ nestedVars, combinedTriggers, Implies(newCond, nestedRhs)(rhs.pos, rhs.info, rhs.errT))(source.pos,MakeInfoPair(source.info, nested.info),MakeTrafoPair(source.errT,nested.errT)))
 
+          case lt@Let(v, e, bod) => {
+            val forallWithoutLet = Forall(vars, triggers, Implies(cond, bod)(rhs.pos, rhs.info))(source.pos, source.info)
+            // desugar the let-body
+            val desugaredWithoutLet = desugarSourceQuantifiedPermissionSyntax(forallWithoutLet)
+            desugaredWithoutLet.map{
+              case SourceQuantifiedPermissionAssertion(iqp, Implies(icond, irhs)) if (!irhs.isPure) =>
+                // Since the rhs cannot be a let-binding, we expand the let-expression
+                Forall(iqp.variables, iqp.triggers.map(t => t.replace(v.localVar, e)), Implies(icond, irhs.replace(v.localVar, e))(iqp.pos, iqp.info))(iqp.pos, iqp.info)

```

Comment:
> // Since the rhs cannot be a let-binding, ...

Why is this the case?

> iqp.triggers.map(t => t.replace(v.localVar, e))

Is this guaranteed to generate valid triggers?
(On a second thought, maybe not, but this may not be a problem because it was the user who chose to use the let-variable in triggers)